VEDA_OS01/views.py

```diff
@@ -96,6 +102,89 @@ def perform_create(self, serializer):
         serializer.save()
 
 
+class TranscriptCredentialsView(APIView):
+    """
+    A Transcript credentials View, used by platform to create/update transcript credentials.
+    """
+
+    def get_cielo_token_response(self, username, api_secure_key):
+        """
+        Returns Cielo24 api token.
+
+        Arguments:
+            username(str): Cielo24 username
+            api_securekey(str): Cielo24 api key
+
+        Returns:
+            Response : Http response object
+        """
+        return requests.get(CIELO24_LOGIN_URL, params={
+            'v': CIELO24_API_VERSION,
+            'username': username,
+            'securekey': api_secure_key
+        })
+
+    def post(self, request):
+        """
+        Creates or updates the org-specific transcript credentials with the given information.
+
+        Arguments:
+            request: A WSGI request.
+        """
+        # Validate invalid provider
+        provider = request.data.get('provider')
```

Comment:
Can we move general validations into another method and make this endpoint more comprehensive. With that you can unit test the validation module separately.